### ISSUE TITLE
Type vs Name resolution breaks when type is a nullable value type

### ISSUE THREAD

**Issue by mqudsi** [dotnet/roslyn#57710]
*I don't know if this behavior is intentional to address some corner case I'm not seeing; if it is, please forgive this erroneous bug report*

Changing only the type of a property from a simple struct to a nullable version of the same struct either changes (if this isn't a bug) or breaks (if this is a bug) same-name type resolution. This is inconsistent with the behavior observed when the property is a regular value type, a reference type, or a nullable reference type.

**Version Used**: 

Visual Studio 17.0.0 Preview 5.0 with a C# 10 project.

**Steps to Reproduce**:

```csharp
public class Foo {
    public TimeSpan? TimeSpan => TimeSpan.FromDays(7);
}
```

**Expected Behavior**:

The compiler will be able to deduce that in `TimeSpan.FromDays()` the token `TimeSpan` refers to a type and not the instance by the same name.

**Actual Behavior**:

An error is observed as `TimeSpan` is resolved to the property by that name and not the type with that name, and an instance of `TimeSpan` does not have a `FromDays()` function.

Note that this behavior only occurs for a nullable value type, and is not observed with any of the other possible combinations. The following all compile OK:

```csharp
public class Foo {
    public TimeSpan TimeSpan => TimeSpan.FromDays(7);
    public string String => String.IsNullOrEmpty(null) ? "foo" : "bar";
    public Array? Array => Array.BinarySearch(new object[] { }, null) == 0 ? new object[] { } : new object[] { };
}
```

The above are all able to correctly resolve a non-namespaced reference to a type by the same name as the property to the type (yielding access to the static method within).

**Comment by CyrusNajmabadi** [comment]
I don't believe this is a bug based on the following rule:  https://github.com/dotnet/csharplang/blob/main/spec/expressions.md#identical-simple-names-and-type-names

> In a member access of the form E.I, if E is a single identifier, and if the meaning of E as a simple_name (Simple names) is a constant, field, property, local variable, or parameter with the same type as the meaning of E as a type_name (Namespace and type names), then both possible meanings of E are permitted

In this case, the meaning of E as a simple name is a property whose type is `Nullable<TimeSpan>`.  However, the meaning of E as a type name is `TimeSpan`.  These are not the same types, and as such this rule does not apply.

You could always request that this rule *see through* nullable value types, but htat would be something to open a discussion on over at github.com/dotnet/csharplang.  Thanks!

**Comment by CyrusNajmabadi** [comment]
To clarify: the Roslyn repo is for the compiler *implementation*.  If the compiler is following the spec, there is no bug here and this repo is not the right location.   Instead, what you'd want to do is open a discussion which might get the language spec changed to support your use case.  If that was done (no promises on that! :)), then the Roslyn compiler (here) would be updated accordingly.

Thanks!

**Comment by mqudsi** [comment]
I understand completely; thanks for clarifying that the current behavior is within-spec - and most of all, thanks for the quick and detailed reply!

**Comment by CyrusNajmabadi** [comment]
Happy to!  Note: questions here are always welcome, so feel free to keep them coming.  However, you might also find it useful to visit our discord at discord.gg/csharp where you can communicate with team members and passionate community members on these sorts of topics.  Feel free to use whichever you prefer :)
